lib/src/repo.rs

```diff
@@ -113,10 +113,16 @@ impl<'a> RepoRef<'a> {
         }
     }
 
-    pub fn shortest_unique_id_prefix_len(&self, target_id_bytes: &[u8]) -> usize {
+    pub fn shortest_unique_commit_id_prefix_len(&self, target_id: &CommitId) -> usize {
         match self {
-            RepoRef::Readonly(repo) => repo.shortest_unique_id_prefix_len(target_id_bytes),
-            RepoRef::Mutable(_) => target_id_bytes.len() * 2, // TODO
+            RepoRef::Readonly(repo) => repo.shortest_unique_commit_id_prefix_len(target_id),
+            RepoRef::Mutable(_) => target_id.as_bytes().len() * 2, // TODO
+        }
```

Comment:
nit: can be forwarded to `self.index().shortest_unique_commit_id_prefix_len()`.

And the proxy function could be removed? If we add `revisions.disambiguatewithin` feature, in-memory cache might be hosted by `repo`, but I don't know.